File: 10 - Subprograms/blockScopeDemo.java
public class blockScopeDemo {
    private static int x;

    public static void main(String [] args) {
        x = 5;

        {
            int x = 7;
            int y = 12;
            System.out.printf("Inside the block x=%d\n", x);
            System.out.printf("Inside the block y=%d\n", y);
        }

        System.out.printf("Outside the block x=%d\n", x);
        // You can't do this: System.out.printf("Outside the block y=%d\n", y);
        // y is not defined in this lexicographical scope
    }
}

File: 10 - Subprograms/blockScopeDemo.cpp
#include <iostream>

using namespace std;

int main() {
    int x = 5;

    {
        int x;
        int y = 12;
        x = 7;
        cout << "Inside the block x=" << x << endl;
        cout << "Inside the block y=" << y << endl;
    }

    cout << "Outside the block x=" << x << endl;
    // You can't do this: cout << "Outside the block y=" << y << endl;
    // y is not defined the current lexical scope
}